diff --git a/src/amd/common/ac_gpu_info.c b/src/amd/common/ac_gpu_info.c
index 8e4c7af..7ea3662 100644
--- a/src/amd/common/ac_gpu_info.c
+++ b/src/amd/common/ac_gpu_info.c
@@ -276,7 +276,7 @@ ac_fill_compiler_info(struct radeon_info *info, const struct drm_amdgpu_info_dev
       out->sgpr_alloc_granularity = 8;
    }
 
-   if (device_info && device_info->num_shader_visible_vgprs) {
+   if (info->drm_minor >= 45 && device_info && device_info->num_shader_visible_vgprs) {
       /* The Gfx10 VGPR count is in Wave32, so divide it by 2 for Wave64.
        * Gfx6-9 numbers are in Wave64. CDNA also includes Accumulation VGPRs.
        */
@@ -458,7 +458,7 @@ ac_fill_hw_ip_info(struct radeon_info *info, const struct drm_amdgpu_info_device
    }
 
    /* Gfx6-8 don't set ip_discovery_version. */
-   if (ip_info->ip_discovery_version) {
+   if (info->drm_minor >= 48 && ip_info->ip_discovery_version) {
       info->ip[ip_type].ver_major = (ip_info->ip_discovery_version >> 16) & 0xff;
       info->ip[ip_type].ver_minor = (ip_info->ip_discovery_version >> 8) & 0xff;
       info->ip[ip_type].ver_rev = ip_info->ip_discovery_version & 0xff;
@@ -1118,7 +1118,8 @@ void ac_fill_hw_info(struct radeon_info *info, const struct drm_amdgpu_info_devi
    info->max_sa_per_se = device_info->num_shader_arrays_per_engine;
    info->num_cu_per_sh = device_info->num_cu_per_sh;
    info->enabled_rb_mask = device_info->enabled_rb_pipes_mask;
-   info->enabled_rb_mask |= (uint64_t)device_info->enabled_rb_pipes_mask_hi << 32;
+   if (info->drm_minor >= 52)
+      info->enabled_rb_mask |= (uint64_t)device_info->enabled_rb_pipes_mask_hi << 32;
 
    info->memory_freq_mhz_effective *= ac_memory_ops_per_clock(info->vram_type);
 
@@ -1402,9 +1403,9 @@ ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    assert(info->drm_major == 3);
    info->is_amdgpu = true;
 
-   if (info->drm_minor < 54) {
+   if (info->drm_minor < 35) {
       fprintf(stderr, "amdgpu: DRM version is %u.%u.%u, but this driver is "
-                      "only compatible with 3.54.0 (kernel 6.6+) or later.\n",
+                      "only compatible with 3.35.0 (kernel 4.20+) or later.\n",
               info->drm_major, info->drm_minor, info->drm_patchlevel);
       return AC_QUERY_GPU_INFO_FAIL;
    }
@@ -1549,7 +1550,8 @@ ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
     * with AMDGPU_GEM_CREATE_DISCARDABLE.
     */
    info->discardable_allows_big_page = info->gfx_level >= GFX10_3 && info->gfx_level < GFX12 &&
-                                       info->has_dedicated_vram;
+                                       info->has_dedicated_vram &&
+                                       info->drm_minor >= 47;
 
    /* Compute the scratch WAVESIZE granularity in bytes. */
    info->scratch_wavesize_granularity_shift = info->gfx_level >= GFX11 ? 8 : 10;
@@ -1568,6 +1570,7 @@ ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    info->max_gflops = (info->gfx_level >= GFX11 ? 256 : 128) * info->num_cu * info->max_gpu_freq_mhz / 1000;
    info->memory_bandwidth_gbps = DIV_ROUND_UP(info->memory_freq_mhz_effective * info->memory_bus_width / 8, 1000);
 
+   info->has_pcie_bandwidth_info = info->drm_minor >= 51;
    info->pcie_gen = device_info.pcie_gen;
    info->pcie_num_lanes = device_info.pcie_num_lanes;
 
diff --git a/src/amd/common/ac_gpu_info.h b/src/amd/common/ac_gpu_info.h
index be59bc9..96762f5 100644
--- a/src/amd/common/ac_gpu_info.h
+++ b/src/amd/common/ac_gpu_info.h
@@ -387,6 +387,7 @@ struct radeon_info {
    bool has_gpuvm_fault_query;
    /* Whether SR-IOV is enabled or amdgpu.mcbp=1 was set on the kernel command line. */
    bool has_kernelq_reg_shadowing;
+   bool has_pcie_bandwidth_info;
    bool has_default_zerovram_support;
    bool has_tmz_support;
    bool has_trap_handler_support;
diff --git a/src/amd/common/ac_surface.c b/src/amd/common/ac_surface.c
index ed9dfec..7dd6b8f 100644
--- a/src/amd/common/ac_surface.c
+++ b/src/amd/common/ac_surface.c
@@ -2100,6 +2100,9 @@ static bool gfx10_DCN_requires_independent_64B_blocks(const struct radeon_info *
    assert(info->gfx_level >= GFX10);
 
    /* For 4K, DCN requires INDEPENDENT_64B_BLOCKS = 1 and MAX_COMPRESSED_BLOCK_SIZE = 64B. */
+   if (info->drm_minor <= 43)
+      return true;
+
    return config->info.width > 2560 || config->info.height > 2560;
 }
 
diff --git a/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_bo.c b/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_bo.c
index 09d4cab..afa89fd 100644
--- a/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_bo.c
+++ b/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_bo.c
@@ -499,7 +499,7 @@ radv_amdgpu_winsys_bo_create(struct radeon_winsys *_ws, uint64_t size, unsigned
          request.flags |= AMDGPU_GEM_CREATE_VRAM_CLEARED;
    }
 
-   if (flags & RADEON_FLAG_DISCARDABLE)
+   if (flags & RADEON_FLAG_DISCARDABLE && ws->info.drm_minor >= 47)
       request.flags |= AMDGPU_GEM_CREATE_DISCARDABLE;
 
    if (flags & RADEON_FLAG_GFX12_ALLOW_DCC && ws->info.drm_minor >= 58) {
diff --git a/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_cs.c b/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_cs.c
index 86486b9..64d2843 100644
--- a/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_cs.c
+++ b/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_cs.c
@@ -211,7 +211,8 @@ radv_amdgpu_cs_domain(const struct radeon_winsys *_ws)
     * If there is no PCIe info, assume there is enough bandwidth.
     */
    const uint32_t bandwidth_mbps_threshold = 8 * 0.985 * 1024;
-   bool enough_bandwidth = ws->info.pcie_bandwidth_mbps >= bandwidth_mbps_threshold;
+   bool enough_bandwidth =
+      !ws->info.has_pcie_bandwidth_info || ws->info.pcie_bandwidth_mbps >= bandwidth_mbps_threshold;
 
    bool use_sam =
       (enough_vram && enough_bandwidth && ws->info.has_dedicated_vram && !(ws->perftest & RADV_PERFTEST_NO_SAM)) ||
